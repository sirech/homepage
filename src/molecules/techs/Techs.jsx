import React from 'react'

import { Card, CardHeader, CardText, CardBlock, Badge } from 'reactstrap'

const techs = [
  ['Ruby on Rails', 'primary'],
  ['JavaScript', 'info'],
  ['React', 'success'],
  ['SQL', 'info'],
  ['REST', 'info'],
  ['Docker', 'success'],
  ['HTML', 'info'],
  ['CSS', 'info'],
  ['Python', 'warning'],
  ['Java', 'warning'],
  ['jQuery', 'danger']
]

const renderTechs = () => (
  techs.map(([name, color]) => {
    return (
      <Badge key={name} className='mr-1' pill color={color}>{name}</Badge>
    )
  })
)

const Techs = () => (
  <Card id='techs'>
    <CardHeader tag='h5'>Technologies</CardHeader>
    <CardBlock>
      <CardText>Following the <a href='https://www.amazon.com/Pragmatic-Programmer-Journeyman-Master/dp/020161622X'>Pragmatic Programmer</a> advice, I try to learn some new technology every year. My github account is full of small projects I have started to teach myself new things, such as this page itself. A list of them, in no particular order:</CardText>
      {renderTechs()}
    </CardBlock>
  </Card>
)

export default Techs
