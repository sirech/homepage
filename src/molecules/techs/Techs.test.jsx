import React from 'react'
import renderer from 'react-test-renderer'
import Techs from './Techs'

describe('components', () => {
  describe('Techs', () => {
    it('renders correctly', () => {
      const component = renderer.create(<Techs />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
