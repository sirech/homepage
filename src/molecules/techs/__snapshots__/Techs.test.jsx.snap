// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Techs renders correctly 1`] = `
<div
  className="mb-3 card"
  id="techs"
>
  <h5
    className="card-header"
  >
    Technologies
  </h5>
  <div
    className="card-block"
  >
    <p
      className="card-text"
    >
      Following the 
      <a
        href="https://www.amazon.com/Pragmatic-Programmer-Journeyman-Master/dp/020161622X"
      >
        Pragmatic Programmer
      </a>
       advice, I try to learn some new technology every year. My github account is full of small projects I have started to teach myself new things, such as this page itself. A list of them, in no particular order:
    </p>
    <span
      className="mr-1 badge badge-primary badge-pill"
    >
      Ruby on Rails
    </span>
    <span
      className="mr-1 badge badge-info badge-pill"
    >
      JavaScript
    </span>
    <span
      className="mr-1 badge badge-success badge-pill"
    >
      React
    </span>
    <span
      className="mr-1 badge badge-info badge-pill"
    >
      SQL
    </span>
    <span
      className="mr-1 badge badge-info badge-pill"
    >
      REST
    </span>
    <span
      className="mr-1 badge badge-success badge-pill"
    >
      Docker
    </span>
    <span
      className="mr-1 badge badge-info badge-pill"
    >
      HTML
    </span>
    <span
      className="mr-1 badge badge-info badge-pill"
    >
      CSS
    </span>
    <span
      className="mr-1 badge badge-warning badge-pill"
    >
      Python
    </span>
    <span
      className="mr-1 badge badge-warning badge-pill"
    >
      Java
    </span>
    <span
      className="mr-1 badge badge-danger badge-pill"
    >
      jQuery
    </span>
  </div>
</div>
`;
